internal/pkg/exp/umount.go

```diff
@@ -0,0 +1,143 @@
+
+// ErrNotSquashfuse is the error returned when attempting to unmount a path that
+// is not a squashfuse mount.
+var ErrNotSquashfuse = errors.New("not a squashfuse mount")
+
+// ErrBadMountInfo is the error returned if we cannot parse /proc/self/mountinfo.
+var ErrBadMountInfo = errors.New("bad mountinfo")
+
+// checkMounted verifies whether mountPath is a current squashfuse mount.
+func checkMounted(mountPath string) error {
+	mountPath, err := filepath.Abs(mountPath)
+	if err != nil {
+		return err
+	}
+
+	mi, err := os.Open("/proc/self/mountinfo")
+	if err != nil {
+		return fmt.Errorf("failed to open %s: %w", mountInfoPath, err)
+	}
+	defer mi.Close()
+
+	scanner := bufio.NewScanner(mi)
+	for scanner.Scan() {
+		fields := strings.Split(scanner.Text(), " ")
+		if len(fields) < 10 {
+			return fmt.Errorf("%w: not enough fields", ErrBadMountInfo)
+		}
+		//nolint:lll
+		// 1348 63 0:77 / /tmp/siftool-mount-956028386 ro,nosuid,nodev,relatime shared:646 - fuse.squashfuse squashfuse ro,user_id=1000,group_id=100
+		mntTarget := fields[4]
+		// Number of fields is not fixed - so loop over field 7+
+		if mntTarget == mountPath {
+			for _, v := range fields[6:] {
+				if v == "squashfuse" {
+					return nil
+				}
+			}
+			return ErrNotSquashfuse
+		}
+	}
+	return ErrNotMounted
+}
+
+// umountSquashFS unmounts the filesystem at mountPath.
+func umountSquashFS(ctx context.Context, mountPath string, uo umountOpts) error {
+	if err := checkMounted(mountPath); err != nil {
+		return err
+	}
+
+	args := []string{
+		"-u",
+		filepath.Clean(mountPath),
+	}
+	cmd := exec.CommandContext(ctx, uo.fusermountPath, args...) //nolint:gosec
+	cmd.Stdout = uo.stdout
+	cmd.Stderr = uo.stderr
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to umount: %w", err)
+	}
+
+	return nil
+}
+
+// mountOpts accumulates mount options.
+type umountOpts struct {
+	stdout         io.Writer
+	stderr         io.Writer
+	fusermountPath string
+}
+
+// MountOpt are used to specify mount options.
+type UmountOpt func(*umountOpts) error
+
+// OptMountStdout writes standard output to w.
+func OptUmountStdout(w io.Writer) UmountOpt {
+	return func(mo *umountOpts) error {
+		mo.stdout = w
+		return nil
+	}
+}
+
+// OptMountStderr writes standard error to w.
+func OptUmountStderr(w io.Writer) UmountOpt {
+	return func(mo *umountOpts) error {
+		mo.stderr = w
+		return nil
+	}
+}
+
+// OptMountFusermountPath sets the path to the fusermount binary.
+func OptUmountFusermount(path string) UmountOpt {
+	return func(mo *umountOpts) error {
+		mo.fusermountPath = path
+		return nil
+	}
+}
+
+// Umount unmounts the FUSE mounted filesystem at mountPath.
+//
+// Umount may start one or more underlying processes. By default, stdout and stderr of these
+// processes is discarded. To modify this behavior, consider using OptUmountStdout and/or
+// OptUmountStderr.
+func Umount(ctx context.Context, mountPath string, opts ...UmountOpt) error {
```

Comment:
Similar to mount, we should call out `OptUmountFusermountPath` option and the default (lookup from `PATH`).